File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
    int count = 0, maxEl = nums[0];   

        for (int i = 0; i < nums.length; i++) 
        {
            if (count == 0) 
            {
                maxEl = nums[i];
                count++;
            } 
            else if (maxEl == nums[i]) 
            {
                count++;
            } 
            else 
            {
                count--;
            }
        }
        return maxEl;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count =0;
        int candidate = nums[0];
        for(int i: nums){
            if(candidate == i){
                count++;
            }
            else{
                if(count ==0)    candidate = i;
                else{
                    count--;}
            }
        }
        return candidate;
    }
};